### PR TITLE
Fix invalid return type in docblock

### PR DESCRIPTION
**Pull request by dagio**

### CODE REVIEW

File: lib/Doctrine/ORM/QueryBuilder.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ public function expr()
      *
      * @param boolean $cacheable
      *
-     * @return \Doctrine\ORM\AbstractQuery This query instance.
+     * @return QueryBuilder
```
Comment (Koc) [inline lib/Doctrine/ORM/QueryBuilder.php]: It is better use `@return $this` here. An explanation https://github.com/php-fig/fig-standards/pull/169/files#diff-9a10a11696dc38209c125ea9c57a565fR2195

**Comment by DHager** [comment]
For "fluent" chaining, can't you use `@returns self` , or is that only recognized by certain IDEs?

**Comment by dag-io** [comment]
@DHager Yes. For the moment I just fixed the one that were incorrect so know all method returning $this have QueryBuilder as return type so in IDEs we have no more issues reported. `@returns $this ` and `@returns self` are not used a lot in the project so I would let you guys decide the right choice.

**Comment by dag-io** [comment]
@Koc @DHager I updated in favor of `@return $this` as @Koc as recommended in the draft of PSR-5

**Comment by Koc** [comment]
@dag-io can you check consistency in other classes, like https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/AbstractQuery.php and others?

**Comment by dag-io** [comment]
@Koc, it's all good now

**Comment by TomasVotruba** [comment]
:+1:

File: lib/Doctrine/ORM/AbstractQuery.php [hunk 2]
```diff
@@ -183,7 +183,7 @@ public function __construct(EntityManagerInterface $em)
      *
      * @param boolean $cacheable
      *
-     * @return static This query instance.
+     * @return $this This query instance.
```
Comment (Ocramius) [inline lib/Doctrine/ORM/AbstractQuery.php]: `static` was correct here

**Comment by Ocramius** [comment]
I manually applied the patch after a rebase and reverting `$this` to `self` or `static` (depending on context).

Thanks @dag-io, sorry for the long delay!

**Comment by TomasVotruba** [comment]
Thank you @Ocramius

**Pull request closed without merging**